storage/allocator.go

```diff
@@ -198,15 +200,11 @@ func (a *allocator) allocateTargetInternal(required proto.Attributes, existing [
 		if leastStore != nil {
 			return leastStore, nil
 		}
-
-		// Otherwise, we have not found a store. Because more redundancy
-		// is better than less, if relaxConstraints, we still try to find
-		// a target by relaxing an attribute constraint, from last
-		// attribute to first.
-		if len(attrs) == 0 || !relaxConstraints {
+		if len(attrs) == 0 {
+			return nil, util.Errorf("unable to allocate a target store; no candidates available")
+		} else if !relaxConstraints {
 			return nil, util.Errorf("unable to allocate a target store for %s", required)
```

Comment:
order does kinda matter because of how attributes get stripped off one at a time. I'll do the ordered thing